src/leveled_runner.erl

```diff
@@ -630,24 +634,23 @@ accumulate_objects(FoldObjectsFun, InkerClone, Tag, DeferredFetch) ->
                     {B0, K0, _T0} ->
                         {B0, K0}
                 end,
-            JK = {leveled_codec:to_ledgerkey(B, K, Tag), SQN},
+            JK = {leveled_codec:to_objectkey(B, K, Tag), SQN},
             case DeferredFetch of
-                {true, JournalCheck} ->
+                {true, JournalCheck} when MD =/= null ->
```

Comment:
I am not 100% sure this forcing of MD to be not `null` results in the same behaviour when `InJournal` turns out to be false.

Reply:
The type of object that will be found in accumulate_objects should always be something with metadata - only index entries should not have metadata.  The Tag restricts the object to being a head-only object or an actual object - and in both cases they have metadata.

If there is a deferred fetch the metadata is required.  If there is not a deferred fetch, the metadata is in the value that will be fetched so the metadata is not explicitly required.